cluster/plan.go

```diff
@@ -119,34 +120,36 @@ func (c *Cluster) BuildKubeAPIProcess(prefixPath string) v3.Process {
-		"service-account-key-file":           pki.GetKeyPath(pki.ServiceAccountTokenKeyName),
-		"etcd-cafile":                        etcdCAClientCert,
-		"etcd-certfile":                      etcdClientCert,
-		"etcd-keyfile":                       etcdClientKey,
-		"etcd-servers":                       etcdConnectionString,
-		"etcd-prefix":                        etcdPathPrefix,
-		"requestheader-client-ca-file":       pki.GetCertPath(pki.RequestHeaderCACertName),
-		"requestheader-allowed-names":        pki.APIProxyClientCertName,
-		"proxy-client-key-file":              pki.GetKeyPath(pki.APIProxyClientCertName),
-		"proxy-client-cert-file":             pki.GetCertPath(pki.APIProxyClientCertName),
-		"requestheader-extra-headers-prefix": "X-Remote-Extra-",
-		"requestheader-group-headers":        "X-Remote-Group",
-		"requestheader-username-headers":     "X-Remote-User",
+		"insecure-bind-address":                    "127.0.0.1",
+		"bind-address":                             "0.0.0.0",
+		"insecure-port":                            "0",
+		"secure-port":                              "6443",
+		"cloud-provider":                           c.CloudProvider.Name,
+		"allow-privileged":                         "true",
+		"kubelet-preferred-address-types":          "InternalIP,ExternalIP,Hostname",
+		"service-cluster-ip-range":                 c.Services.KubeAPI.ServiceClusterIPRange,
+		"service-node-port-range":                  c.Services.KubeAPI.ServiceNodePortRange,
+		"storage-backend":                          "etcd3",
+		"client-ca-file":                           pki.GetCertPath(pki.CACertName),
+		"tls-cert-file":                            pki.GetCertPath(pki.KubeAPICertName),
+		"tls-private-key-file":                     pki.GetKeyPath(pki.KubeAPICertName),
+		"kubelet-client-certificate":               pki.GetCertPath(pki.KubeAPICertName),
+		"kubelet-client-key":                       pki.GetKeyPath(pki.KubeAPICertName),
+		"service-account-key-file":                 pki.GetKeyPath(pki.ServiceAccountTokenKeyName),
+		"etcd-cafile":                              etcdCAClientCert,
+		"etcd-certfile":                            etcdClientCert,
+		"etcd-keyfile":                             etcdClientKey,
+		"etcd-servers":                             etcdConnectionString,
+		"etcd-prefix":                              etcdPathPrefix,
+		"requestheader-client-ca-file":             pki.GetCertPath(pki.RequestHeaderCACertName),
+		"requestheader-allowed-names":              pki.APIProxyClientCertName,
+		"proxy-client-key-file":                    pki.GetKeyPath(pki.APIProxyClientCertName),
+		"proxy-client-cert-file":                   pki.GetCertPath(pki.APIProxyClientCertName),
+		"requestheader-extra-headers-prefix":       "X-Remote-Extra-",
+		"requestheader-group-headers":              "X-Remote-Group",
+		"requestheader-username-headers":           "X-Remote-User",
+		"authentication-token-webhook-config-file": AuthnWebhookConfigFile,
```

Comment:
I don't think we should allow the webhook token by default, user must have the choice not to use it.

Reply:
Logic has been modified to conditionally add webhook params if config is defined.